src/libpriv/rpmostree-postprocess.c

```diff
@@ -512,34 +512,40 @@ rpmostree_prepare_rootfs_get_sepolicy (int            dfd,
   return ret;
 }
 
-static gboolean
-replace_nsswitch (int            dfd,
-                  GCancellable  *cancellable,
-                  GError       **error)
+char *
+rpmostree_postprocess_replace_nsswitch_string (const char *buf,
+                                               GError    **error)
 {
-  g_autofree char *nsswitch_contents = NULL;
-  g_autofree char *new_nsswitch_contents = NULL;
-
   static gsize regex_initialized;
   static GRegex *passwd_regex;
 
   if (g_once_init_enter (&regex_initialized))
     {
-      passwd_regex = g_regex_new ("^(passwd|group):\\s+files(.*)$",
+      passwd_regex = g_regex_new ("^(passwd|group):\\s+(.*)files(.*)$",
```

Comment:
Hmm, this doesn't catch cases where an entry contains `files` in its name.

Maybe something like: `\s+((?:[\S]* )*)files((?: [\S]*)*)$`? (not tested). Or maybe at this point, it's just easier to extract what's after the `\s+`, split that ourselves, insert `altfiles` and rejoin.